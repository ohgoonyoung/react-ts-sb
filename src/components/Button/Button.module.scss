@import "src/styles/scss/common.scss";

.big {
  width : 200px;
}
.small {
  width : 100px;
}
.button {
  height : 39px;
  border-radius: 2;
}

.primary {
  background-color:$primary-color;
  color:$white-color;
}
.secondary {
  background-color:$secondary-color;
  color:$white-color;
}

.brownishGrey {
  background-color:$brownishGrey-color;
  color:$white-color;
}

.bordered {
  background-color:$white-color !important;
  color:$black-color;
  &.primary {
    border : 1px solid $primary-color !important;
  }
  &.secondary {
    border : 1px solid $secondary-color !important;
  }
  &.brownishGrey {
    border : 1px solid $brownishGrey-color !important;
  }
}