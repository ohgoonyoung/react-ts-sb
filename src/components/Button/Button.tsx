import * as React from 'react';
import defaultStyle from './Button.module.scss';
import cn from 'classnames';

type SizeTypes = "big" | "small";
type colorTypes = "default" | "primary" | "secondary" | "brownishGrey"

interface IProps {
  size: SizeTypes;
  color: colorTypes;
  bordered?: boolean;
}

const Button: React.FC<IProps> = ({ size, color, bordered, children }) => {
  return (
    <button className={cn(defaultStyle.button, defaultStyle[size], bordered ? defaultStyle.bordered : null, defaultStyle[color])}>{children}</button>
  );
}

Button.defaultProps = {
  size: "big"
};

export default Button;
