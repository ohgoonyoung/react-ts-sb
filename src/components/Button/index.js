export default Button from './Button';