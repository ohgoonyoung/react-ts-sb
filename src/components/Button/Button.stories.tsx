import React from 'react';

import { storiesOf } from '@storybook/react';
import Button from './Button';

storiesOf('Button', module)
  .add('Button Theme', () =>
    <div>
      {/* <div style={{ display: "flex", padding: 10 }}>
        <Button size="big" color="default" >default</Button>
      </div> */}
      <div style={{ padding: 10 }}>
        <Button size="big" color="primary">primary</Button>
      </div>
      <div style={{ padding: 10 }}>
        <Button size="big" color="secondary">secondary</Button>
      </div>
      <div style={{ padding: 10 }}>
        <Button size="big" color="brownishGrey">brownishGrey</Button>
      </div>
    </div>)

  .add('Outline Button', () =>
    <div>
      <div style={{ padding: 10 }}>
        <Button bordered size="big" color="primary" >primary</Button>
      </div>
      <div style={{ padding: 10 }}>
        <Button bordered size="big" color="secondary" >secondary</Button>
      </div>
      <div style={{ padding: 10 }}>
        <Button bordered size="big" color="brownishGrey" >brownishGrey</Button>
      </div>
    </div>);