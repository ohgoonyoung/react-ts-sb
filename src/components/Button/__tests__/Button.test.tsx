import * as React from "react";
import { mount } from 'enzyme';
import Button from 'src/components/Button/Button';

describe('Button', () => {
  it('Button Theme', () => {
    const wrapper = mount(<Button size="big" color="primary">primary</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('Outline Button', () => {
    const wrapper = mount(<Button bordered size="big" color="primary">primary</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});