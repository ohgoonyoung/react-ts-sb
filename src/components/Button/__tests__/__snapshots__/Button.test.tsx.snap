// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button Theme 1`] = `ReactWrapper {}`;

exports[`Button Outline Button 1`] = `ReactWrapper {}`;
